tests/integration/routes/test_questionnaire.py

```diff
@@ -1,24 +1,31 @@
-import simplejson as json
-from mock import Mock
-
-from app.data_models import QuestionnaireStore
 from tests.integration.integration_test_case import IntegrationTestCase
 
 
 class TestQuestionnaire(IntegrationTestCase):
-    def setUp(self):
-        super().setUp()
-        self._application_context = self._application.app_context()
-        self._application_context.push()
-
-        storage = Mock()
-        data = {"METADATA": "test", "ANSWERS": [], "PROGRESS": []}
-        storage.get_user_data = Mock(
-            return_value=(json.dumps(data), QuestionnaireStore.LATEST_VERSION)
+    def test_head_request_on_optional_date(self):
```

Comment:
Happy to keep as is, but this test would only ensure `block` route is valid. Ideally, we should make a HEAD request for each of our route definitions.

Reply:
I've added tests for the additional questionnaire endpoints. This highlighted that there was a gap in the logic for the session endpoints, so I've fixed that and added tests.